Framework/Algorithms/src/ReflectometrySumInQ.cpp

```diff
@@ -504,19 +531,35 @@ ReflectometrySumInQ::projectedLambdaRange(const MinMax &wavelengthRange,
  */
 ReflectometrySumInQ::Angles
 ReflectometrySumInQ::referenceAngles(const API::SpectrumInfo &spectrumInfo) {
+  const auto beamCentreProperty = getPointerToProperty(Prop::BEAM_CENTRE);
+  const auto thetaInProperty = getPointerToProperty(Prop::THETA_IN);
   Angles a;
-  const int beamCentre = getProperty(Prop::BEAM_CENTRE);
-  const double centreTwoTheta =
-      spectrumInfo.twoTheta(static_cast<size_t>(beamCentre));
-  const bool isFlat = getProperty(Prop::IS_FLAT_SAMPLE);
-  if (isFlat) {
-    a.horizon = centreTwoTheta / 2.;
+  // If BeamCentre is defined use that else use ThetaIn
+  if (!beamCentreProperty->isDefault()) {
+    const int beamCentre = getProperty(Prop::BEAM_CENTRE);
+    const double centreTwoTheta =
+        spectrumInfo.twoTheta(static_cast<size_t>(beamCentre));
+    const bool isFlat = getProperty(Prop::IS_FLAT_SAMPLE);
+    if (isFlat) {
+      a.horizon = centreTwoTheta / 2.;
+    } else {
+      a.horizon = 0.;
+    }
+    a.twoTheta = centreTwoTheta;
+    a.delta = a.twoTheta - a.horizon;
+    return a;
+  } else if (!thetaInProperty->isDefault()) {
+    const int thetaIn = getProperty(Prop::THETA_IN);
+    const double twoTheta = spectrumInfo.twoTheta(static_cast<size_t>(thetaIn));
+    a.horizon = thetaIn;
+    a.twoTheta = twoTheta;
+    a.delta = a.twoTheta - a.horizon;
+    return a;
   } else {
-    a.horizon = 0.;
+    throw std::runtime_error("Neither " + Prop::BEAM_CENTRE + " or " +
```

Comment:
This should probably be done in `validateInputs()`.

Reply:
It is done in validateInputs but, I added an else at the end of a chain of if and else if etc, which I think is in the coding standards....